lib/msf/core/post/file.rb

```diff
@@ -712,25 +710,38 @@ def _append_file_powershell(file_name, data)
 
   def _write_file_powershell(file_name, data, append = false)
     offset = 0
-    chunk_size = 16256
+    chunk_size = 1000
     loop do
-      _write_file_powershell_fragment(file_name, data, offset, chunk_size, append)
-      offset += chunk_size + 1
+      success = _write_file_powershell_fragment(file_name, data, offset, chunk_size, append)
+      unless success
+        unless offset == 0
+          vprint_status("Write partially succeeded then failed. May need to manually clean up")
+        end
+        return false
+      end
+
+      # Future writes will then append, regardless of whether this is an append or write operation
+      append = true
+      offset += chunk_size
       break if offset >= data.length
     end
+
+    true
   end
 
   def _write_file_powershell_fragment(file_name, data, offset, chunk_size, append = false)
-    chunk = data[offset..offset + chunk_size]
+    token = ::Rex::Text.rand_text_alpha(32)
```

Comment:
It's a bit of an obscure edgecase, but it might be nice to prefix this with a `_` to ensure that we don't ever overlap with the strict_encode64 characters